R/constructors-Node.R

```diff
@@ -0,0 +1,157 @@
+  return(NodeIdList(getNodeIds(object), uniqueOnly = uniqueOnly))
+})
+
+#' @export
+setMethod("NodeIdList", "character", function(object, uniqueOnly = c(TRUE, FALSE)) {
+  uniqueOnly <- veupathUtils::matchArg(uniqueOnly)
+
+  if (length(object) == 0) {
+    stop("nodeIds must not be empty")
+  }
+
+  if (uniqueOnly) {
+    object <- unique(object)
+  }
+
+  return(new("NodeIdList", S4Vectors::SimpleList(lapply(object, NodeId))))
+}) 
+
+#' @export 
+setMethod("NodeIdList", "data.frame", function(object, uniqueOnly = c(TRUE, FALSE)) {  
+  if (!isValidEdgeList(object)) {
+    stop(paste(errors, collapse = '\n'))
+  }
+
+  return(NodeIdList(c(object$source, object$target), uniqueOnly = uniqueOnly))
+})
+
+#' @export 
+setMethod("NodeIdList", "missing", function(object, uniqueOnly = c(TRUE, FALSE)) {
+  return(new("NodeIdList"))
+})
+
+#' @export 
+setMethod("NodeIdList", "Node", function(object, uniqueOnly = c(TRUE, FALSE)) {
+  return(NodeIdList(list(object)))
+})
+
+
+#' Create a Node
+#' 
+#' @param id string a unique identifier for the node
+#' @param x numeric value indicating the x coordinate of the node. Optional.
+#' @param y numeric value indicating the y coordinate of the node. Optional.
+#' @param color string or numeric that determines the color of the node. Optional.
+#' @param weight numeric value associated with the node, such as timestamp or other node-associated data. Optional.
+#' @export
+setGeneric("Node", function(id, x = numeric(), y = numeric(), color = NULL, weight = NULL) standardGeneric("Node"), signature = c("id"))
+
+#' @export
+setMethod("Node", "numeric", function(id, x = numeric(), y = numeric(), color = NULL, weight = NULL) {
+  new("Node", id = NodeId(as.character(id)), x = x, y = y, color = color, weight = weight)
+})
+
+#' @export
+setMethod("Node", "character", function(id, x = numeric(), y = numeric(), color = NULL, weight = NULL) {
+  new("Node", id = NodeId(id), x = x, y = y, color = color, weight = weight)
+})
+
+#' @export
+setMethod("Node", "NodeId", function(id, x = numeric(), y = numeric(), color = NULL, weight = NULL) {
+  new("Node", id = id, x = x, y = y, color = color, weight = weight)
+})
+
+#' @export 
+setMethod("Node", "missing", function(id, x = numeric(), y = numeric(), color = NULL, weight = NULL) {
+  new("Node", id = NodeId(generate_node_id(1)), x = x, y = y, color = color, weight = weight)
+})
+
+
+#' @include utils.R
+#' Generate a NodeList
+#' 
+#' Generate a NodeList from an edgeList
+#' @param object Object containing data to be converted to a NodeList
+#' @return NodeList
+#' @export
+#' @examples
+#' NodeList(data.frame(source='a',target='b'))
+setGeneric("NodeList", function(object) standardGeneric("NodeList"))
+
+#' @export
+setMethod("NodeList", "data.frame", function(object = data.frame(source=character(),target=character())) {
+  if (!isValidEdgeList(object)) {
+    stop(paste(errors, collapse = '\n'))
```

Comment:
where does `errors` come from? I dont see it initialized here

Reply:
magic! jk lol. i need to fix this up, and write some tests w bad edgeLists. isValidEdgeList should either be true or a list of errors, and i was moving too quick when i wrote this.